File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int calcSum = (n*(n+1))/2;
        int sum = 0;
        for (int i = 0; i < n; i++)
        {
            sum = sum + nums[i];
        }
        int ans = calcSum - sum;
        return ans;
    }
}

File: 268-missing-number/missing-number.cpp
class Solution {
public:
    bool linearSearch(vector<int> &nums , int num) {
        int n = nums.size();
        for(int i = 0; i < n; i++)
        {
            if(nums[i] == num) 
            {
                return true;
            }
            
        }
        return false;
    }
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        for(int i = 0; i < n+1; i++) {
            if(!linearSearch(nums,i))
            {
                return i;
            }
        }
        return -1;
    }
};